### PR TITLE
Add option to hide nonloggedin/nonsubmitted teams on the scoreboard

### PR DESCRIPTION
**Pull request by vmcj**
This is a config option as a config option for the following 2 cases:
- Normally you want a team which has logged on on the scoreboard
- In case of the autologin tool of GEHACK/Nicky the user would also login without interaction of the actual team

I've tested it manually and it works.

### CODE REVIEW

**Comment by vmcj** [comment]
IMO closes: https://github.com/DOMjudge/domjudge/issues/2061

File: etc/db-config.yaml [hunk 1]
```diff
@@ -269,6 +269,15 @@
             default_value: false
             public: true
             description: Show canonical compiler and runner version on the team pages.
+        -   name: hide_teams_scoreboard_condition
```
Comment (meisterT) [inline etc/db-config.yaml:272]: Consider phrasing these in a positive form, e.g. show_teams_on_scoreboard: always, after_login, after_submission

**Review by meisterT — COMMENTED** [review]
I like it especially for online judges /contests with self registration enabled that seems useful.

**Comment by nickygerritsen** [comment]
> This is a config option as a config option for the following 2 cases:
> 
> * Normally you want a team which has logged on on the scoreboard
> * In case of the autologin tool of GEHACK/Nicky the user would also login without interaction of the actual team
> 
> I've tested it manually and it works.

Note that the autologin tool should only set last API login, not last UI login.

**Review by nickygerritsen — APPROVED** [review]

**Comment by vmcj** [comment]
> > This is a config option as a config option for the following 2 cases:
> > 
> > * Normally you want a team which has logged on on the scoreboard
> > * In case of the autologin tool of GEHACK/Nicky the user would also login without interaction of the actual team
> > 
> > I've tested it manually and it works.
> 
> Note that the autologin tool should only set last API login, not last UI login.

I thought so but I was not sure, but still I think people can come up with an usecase for both. I'm fine with making this a boolean but than I need to know which one we prefer.

**Comment by nickygerritsen** [comment]
I think the 3 options are totally fine.

**Comment by vmcj** [comment]
> > This is a config option as a config option for the following 2 cases:
> > 
> > * Normally you want a team which has logged on on the scoreboard
> > * In case of the autologin tool of GEHACK/Nicky the user would also login without interaction of the actual team
> > 
> > I've tested it manually and it works.
> 
> Note that the autologin tool should only set last API login, not last UI login.

As discussed in Slack, I'll also fix the issue where when you use only the submit client I'll still put you on the scoreboard as you did a login via the API (but need to see if I should check for users with !(GUI OR submission) or on both login fields (and get the GEHACK scenario)

**Comment by vmcj** [comment]
I'll fix it tonight and wait for @ubergeek42 if this is indeed a fix to his issue.

**Comment by ubergeek42** [comment]
This sounds like it'd fix the issue we saw at NAC

**Pull request merged**